### PR TITLE
Handle array values for `env` when getting queue

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

**Review by sarahhodne — COMMENTED** [review]
The line is getting quite long and hard to parse, but I'm pretty sure it's doing the right thing, assuming you only want to consider the first `env` entry if there's an array?

**Comment by meatballhat** [comment]
I agree, it's getting pretty silly.  I'd like to take a crack at making this application less silly overall.
